### ISSUE TITLE
macOS Sierra: auto-login and re-login on wakeup does not work

### ISSUE THREAD

**Issue by simonarnell** [iscsi-osx/iSCSIInitiator#66]
On a macOS Sierra initiator to a Synology Diskstation target, I lose my iscsi connection after my Mac goes to sleep and wakes up. To remount I have to manually perform a iscsictl login.

**Comment by nsinenian** [comment]
This is known issue with Sierra. I'm assuming that Apple has made changes to the way the network stack operates when an interfaces comes up. The issue is observed not only wakeup after sleep but also intermittent loss of network connectivity.

At this moment I don't fully understand what Apple has changed, and why this would occur. Stay tuned.

**Comment by nsinenian** [comment]
@simonarnell  Do you mind telling me what model disputation you are using? My request completely unrelated to this issue, but has to do with another issue where a user cannot connect (they are also using a Synology product and I wand to know what's different).

**Comment by simonarnell** [comment]
DS412+ running DSM 6.0.2-8451

**Comment by ryansyah** [comment]
Hello,

I can comment that I am experience the same behavior with macOS sierra as the initiator to a ctld target on FreeBSD.

**Comment by nsinenian** [comment]
@ryansyah Thanks -- due to changes in Sierra (either the networking stack, how sleep is handled or some of the APIs) this functionality is now broken. We're trying to find a good workaround that won't be as easily affected by changes in future releases.

**Comment by nsinenian** [comment]
@ryansyah @simonarnell Please try the attached build. 

@ryansyah It should also solve your other problem (login <target>,<portal>)

[iSCSIInitiator-1.0.0-beta4+.zip](https://github.com/iscsi-osx/iSCSIInitiator/files/545826/iSCSIInitiator-1.0.0-beta4.zip)

**Comment by ryansyah** [comment]
Nareg,

Should I uninstall the old version first, or is just installing over the old version sufficient?

I tried the installing on top of method last night, and was unable to get autologin working. I am assuming that after install of the linked build, reboot system, and that the disk should autologin and mount. 

-- 
Ryan Stark

**Comment by nsinenian** [comment]
I suggest that you remove it, reboot, re-install, reboot.

There's some caching (e.g., kext caching) that occurs and I'm not quite
sure how Sierra handles that now, so that is your best best. I believe that
this new build should re-mount drives after a wakeup from sleep in Sierra,
but let me know.

**Comment by rax-rstark** [comment]
Nareg,

I updated to 10.12.1 on Tuesday. I'll try to remove and complete reinstall tonight and let it ride for 24 hours and report back with my findings.

**Comment by ryansyah** [comment]
Nareg,

I updated to 10.12.1 on Tuesday. I'll try to remove and complete reinstall the iscsi bits tonight per your recommendation, let it ride for 24 hours, and report back with my findings.

**Comment by ryansyah** [comment]
OK, so I have uninstalled, rebooted, reinstalled and rebooted.

I took note of the kext details before and after and they appeared to be the same.

before:
```
123    0 0xffffff7f80b7f000 0xb000     0xb000     com.github.iscsi-osx.iSCSIInitiator (1) 0E05E8B4-5E71-3DFD-9C80-897EA59A3EAF <122 5 4 3 1>
```

after reinstall, reboot:
```
127    0 0xffffff7f80b7f000 0xb000     0xb000     com.github.iscsi-osx.iSCSIInitiator (1) 0E05E8B4-5E71-3DFD-9C80-897EA59A3EAF <126 5 4 3 1>
```
Once rebooted, the iscsi disk share did not automount. I then ran the following commands:
```
iscsictl modify target-config iqn.2003-12.com.example:pictures -auto-login enable
Target settings have been updated

iscsictl login iqn.2003-12.com.example:pictures
Attached iqn.2003-12.com.example:pictures <scsi domain 0, target 0>
	FREEBSD CTLDISK 0001
	Serial Number MYSERIAL   0
	lun 0: type 0x00 (Block device)
```

Once that is done, my disk mounts and is accessible. I then launch Photos, with it's system `Photos Library.photoslibrary` directory living on the iscsi disk and wait to see if anything bad happens.

I should note that I ensure that my computer is on at all times and does not go to sleep. Computer Sleep setting is set to never, however I do note that I had box checked for 'Put hard disks to sleep when possible'. I am thinking this setting may be related, so I've gone ahead and unchecked that setting too. I'll check back in 24 hours to see how things are doing and report back here.

**Comment by ryansyah** [comment]
22 hours of uptime, no panics, no disk connection issues. Photos did complain and had to quit, but it was able to restart cleanly and access the photos library on the mounted iscsi target disk. No further messages in system.log, no new crash dumps. This definitely feels like an improvement.

**Comment by nsinenian** [comment]
@ryansyah Can you please test the attached build below for auto-login at startup? (Change the extension to DMG and then install as usual)

[iSCSIInitiator-1.0.0-beta4+++.zip](https://github.com/iscsi-osx/iSCSIInitiator/files/628787/iSCSIInitiator-1.0.0-beta4.zip)

**Comment by TychoAI** [comment]
I installed beta4+++ and it seems to be mounting both a login and upon wake from sleep. Will keep an eye out for it, but so far this resolves both the Synology "unable to format" issue and the auto-mount. Things are looking good! Thanks, Nareg.

**Comment by ryansyah** [comment]
I too can confirm that autologin at boot is now functioning as expected. I'll play around with this over the weekend to see how things are running and report back.

Thanks Nareg!

**Comment by ryansyah** [comment]
I've had this solution up and running without issue with the Photos application idling in the background for 5 days now. My issues are fully resolved in this build, Thanks!

Now to reboot into 10.12.2 and see if anything changes.
